File: web/e2_viewer/src/grep/gitem.test.ts
import { expect, test } from 'vitest'
import { GPoint, GLine, GPlane, GSphere, GBlock } from './gitem.js'

test ('GPoint', () => {
    const point = new GPoint(5);
    expect(point.size()).toBe(5);
    expect(point.toString()).toBe('GPoint(5)');
});

test('GLine', () => {
    const line = new GLine(10);
    expect(line.length()).toBe(10);
    expect(line.toString()).toBe('GLine(10)');
});

test('GPlane', () => {
    const plane = new GPlane(10, 20);
    expect(plane.width()).toBe(10);
    expect(plane.height()).toBe(20);
    expect(plane.toString()).toBe('GPlane(10, 20)');
});

test('GSphere', () => {
    const sphere = new GSphere(15);
    expect(sphere.radius()).toBe(15);
    expect(sphere.toString()).toBe('GSphere(15)');
});

test('GBlock', () => {
    const block = new GBlock(5, 10, 15);
    expect(block.width()).toBe(5);
    expect(block.height()).toBe(10);
    expect(block.depth()).toBe(15);
    expect(block.toString()).toBe('GBlock(5, 10, 15)');
});

File: engines/src/grep/gitem.test.cpp

#include <gtest/gtest.h>
#include "grep/gitem.h"

using namespace e2;

TEST(GPointTest, BasicFunctionality) {
    GPoint point(5);
    EXPECT_EQ(point.type(), GItemType::GPOINT);
    EXPECT_EQ(point.size(), 5);
    EXPECT_EQ(point.toString(), "GPoint(size=5)");
}

TEST(GLineTest, BasicFunctionality) {
    GLine line(10);
    EXPECT_EQ(line.type(), GItemType::GLINE);
    EXPECT_EQ(line.length(), 10);
    EXPECT_EQ(line.toString(), "GLine(length=10)");
}

TEST(GPlaneTest, BasicFunctionality) {
    GPlane plane(10, 20);
    EXPECT_EQ(plane.type(), GItemType::GPLANE);
    EXPECT_EQ(plane.width(), 10);
    EXPECT_EQ(plane.height(), 20);
    EXPECT_EQ(plane.toString(), "GPlane(width=10, height=20)");
}

TEST(GSphereTest, BasicFunctionality) {
    GSphere sphere(15);
    EXPECT_EQ(sphere.type(), GItemType::GSPHERE);
    EXPECT_EQ(sphere.radius(), 15);
    EXPECT_EQ(sphere.toString(), "GSphere(radius=15)");
}

TEST(GBlockTest, BasicFunctionality) {
    GBlock block(5, 10, 15);
    EXPECT_EQ(block.type(), GItemType::GBLOCK);
    EXPECT_EQ(block.width(), 5);
    EXPECT_EQ(block.height(), 10);
    EXPECT_EQ(block.depth(), 15);
    EXPECT_EQ(block.toString(), "GBlock(width=5, height=10, depth=15)");
}

